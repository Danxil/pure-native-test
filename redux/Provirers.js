import React from 'react';
import { compose, pure } from 'recompose';
import { Provider } from 'react-redux';

import configureStore from './configureStore';

type Provirers = {
  children: React.Node,
}
const Providers = ({ children }: Provirers) => (
  <Provider store={configureStore()}>
    {children}
  </Provider>
);
export default compose(pure)(Providers);
