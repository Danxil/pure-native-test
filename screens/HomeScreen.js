import React, { useEffect } from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';

import user from '../redux/hooks/selectors/user';
import userActions from '../redux/hooks/actions/userActions';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const HomeScreen = () => {
  const userInfo = user();
  const { getUserInfo } = userActions();
  console.log(userInfo);
  useEffect(
    () => {
      getUserInfo();
    },
    [],
  );
  return (
    <View style={styles.container}>
      <Text>Home Screen</Text>
    </View>
  );
};
export default React.memo(HomeScreen);
