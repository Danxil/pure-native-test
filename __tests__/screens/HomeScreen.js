import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import HomeScreen from '../../screens/HomeScreen';
import Provirers from '../../redux/Provirers';

it('renders correctly', () => {
  const comp = shallow(<Provirers><HomeScreen /></Provirers>);
  expect(comp).toMatchSnapshot();
});
