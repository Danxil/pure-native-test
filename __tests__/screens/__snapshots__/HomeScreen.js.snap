// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Providers>
  <ConnectFunction />
</Providers>
`;
